File: 55-jump-game/jump-game.py
class Solution:
    def canJump(self, nums):
        k = 0
        i = 0

        while i < len(nums):

            if i > k:
                return False

            k = max(k, i + nums[i])

            if k >= len(nums) - 1:
                return True

            i += 1

        return True

File: 55-jump-game/jump-game.java
class Solution {
    public boolean canJump(int[] nums) {

        int k = 0;
        int i = 0;

        while (i < nums.length) {

            if (i > k) {
                return false;
            }

            k = Math.max(k, i + nums[i]);

            if (k >= nums.length - 1) {
                return true;
            }

            i++;
        }

        return true;
    }
}